### PR TITLE
chore(rc): Refactor RC integration tests

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Refactor RC integration tests by adding separate test per operation

### CODE REVIEW

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java [hunk 1]
```diff
@@ -42,89 +42,133 @@
...
-    assertNotEquals(VERSION, publishedTemplate.getVersion());
+    assertNotEquals(version, publishedTemplate.getVersion());
+  }
+
+  @Test
+  public void testGetTemplate() throws FirebaseRemoteConfigException {
+    // get template to fetch the active template with correct etag
+    // modify and publish a known template to test get template operation.
+    final Template oldTemplate = remoteConfig.getTemplate();
+    final Template inputTemplate = Template.fromJSON(oldTemplate.toJSON());
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:101]: Are you using this as a way to "clone" the original template? If so might be better to add a `cloneTemplate()` helper method with this logic.

Comment (hiranya911) [reply src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:101]: Also we probably don't need to clone it for this test case?

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java [hunk 2]
```diff
@@ -42,89 +42,133 @@
...
     Template publishedTemplate = remoteConfig.publishTemplate(inputTemplate);
+
     assertNotEquals(inputTemplate.getETag(), publishedTemplate.getETag());
-    assertEquals(PARAMETERS, publishedTemplate.getParameters());
-    assertEquals(PARAMETER_GROUPS, publishedTemplate.getParameterGroups());
+    assertEquals(parameters, publishedTemplate.getParameters());
+    assertEquals(parameterGroups, publishedTemplate.getParameterGroups());
     assertEquals(CONDITIONS, publishedTemplate.getConditions());
-    assertNotEquals(VERSION, publishedTemplate.getVersion());
+    assertNotEquals(version, publishedTemplate.getVersion());
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:93]: Shouldn't they have the same description?

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java [hunk 3]
```diff
@@ -42,89 +42,133 @@
...
+  }
+
+  @Test
+  public void testPublishTemplate() throws FirebaseRemoteConfigException {
     // get template to fetch the active template with correct etag
     final Template oldTemplate = remoteConfig.getTemplate();
     final Template inputTemplate = Template.fromJSON(oldTemplate.toJSON());
-    final String versionNumber = oldTemplate.getVersion().getVersionNumber();
-
+    final Map<String, Parameter> parameters = getParameters();
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:77]: Probably doesn't need to clone the template

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Looks pretty good. Just a few questions to address.

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java (hunk 3 above)
Comment (lahirumaramba) [reply src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:77]: Good catch! Thanks!

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java (hunk 2 above)
Comment (lahirumaramba) [reply src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:93]: They should! Added a new assertEquals for description. Thanks!

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java (hunk 1 above)
Comment (lahirumaramba) [reply src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:101]: Good catch. Thanks! This was only needed in the previous design. Removed the cloning from all the tests.
We only need to copy the etag for `testGetTemplateAtVersion()`. I changed the code there to use the constructor, instead.

**Review by hiranya911 — APPROVED** [review]
Thanks. LGTM 👍

**Pull request merged**